ring/ring.go

```diff
@@ -601,7 +642,7 @@ func (r *Ring) updateRingMetrics() {
 	for state, timestamp := range oldestTimestampByState {
 		r.oldestTimestampGaugeVec.WithLabelValues(state).Set(float64(timestamp))
 	}
-	r.totalTokensGauge.Set(float64(len(r.ringTokens)))
+	r.totalTokensGauge.Set(float64(ringTokenCount))
```

Comment:
Why `len(r.ringTokens)` wasn't good enough? You can read it while the lock is held at the beginning of the function.

Reply:
went with different solution